types/react/test/index.ts

```diff
@@ -459,8 +459,8 @@ const MemoizedForwardingRefComponent = React.memo(ForwardingRefComponent);
 const LazyComponent = React.lazy(() => Promise.resolve({ default: RefComponent }));
 
 type ClassComponentAsRef = React.ElementRef<typeof RefComponent>; // $ExpectType RefComponent
-type FunctionComponentWithoutPropsAsRef = React.ElementRef<typeof RefCarryingComponent>; // $ExpectType undefined
-type FunctionComponentWithPropsAsRef = React.ElementRef<typeof FunctionComponent>; // $ExpectType undefined
+type FunctionComponentWithoutPropsAsRef = React.ElementRef<typeof RefCarryingComponent>; // $ExpectType never
+type FunctionComponentWithPropsAsRef = React.ElementRef<typeof FunctionComponent>; // $ExpectType never
```

Comment:
@alloy Could you take a look at these changes and tell me if I'm missing something? It seems to me that `never` is a more appropriate type for these two cases. The components would reject `ref` anyway. It would never assign `undefined` anyway. The instance would always be `null`.

Reply:
Yeah that seems fine to me–I used `undefined` simply to match the upstream Flow documentation.